examples/quickstart-xgboost/client.py

```diff
@@ -0,0 +1,197 @@
+import warnings
+import argparse
+from logging import INFO
+import xgboost as xgb
+
+import flwr as fl
+from flwr_datasets import FederatedDataset
+from flwr.common.logger import log
+from flwr.common import (
+    Code,
+    EvaluateIns,
+    EvaluateRes,
+    FitIns,
+    FitRes,
+    GetParametersIns,
+    GetParametersRes,
+    Parameters,
+    Status,
+)
+
+from dataset import (
+    instantiate_partitioner,
+    train_test_split,
+    transform_dataset_to_dmatrix,
+)
+
+
+warnings.filterwarnings("ignore", category=UserWarning)
+
+
+def args_parser():
+    """Parse arguments to define experimental settings."""
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--num_partitions", default=20, type=int, help="Number of partitions."
+    )
+    parser.add_argument(
+        "--partitioner_type",
+        default="uniform",
+        type=str,
+        choices=["uniform", "linear", "square", "exponential"],
+        help="Partitioner types.",
+    )
+    parser.add_argument(
+        "--partition_id",
+        default=0,
+        type=int,
+        help="Partition ID used for the current client.",
+    )
+    parser.add_argument(
+        "--seed", default=42, type=int, help="Seed used for train/test splitting."
+    )
+    parser.add_argument(
+        "--test_fraction",
+        default=0.2,
+        type=float,
+        help="test fraction for train/test splitting.",
+    )
+
+    args_ = parser.parse_args()
+    return args_
```

Comment:
That doesn't belong to this file. Maybe utils. (btw it's also poorly added in the advanced-pytorch but let's not repeat it)

Reply:
agree~